### ISSUE TITLE
Support for Flags?

### ISSUE THREAD

**Issue by AlexEndris** [dennisdoomen/fluentassertions#249]
While I am aware that the assertion of enum flags can already be achieved, it would be amazing if there was an extension method (e.g. HaveFlag) to directly support it.

So my question is not about the possibility to include it (I wrote my own extension method for it), but more if there was a specific reason to not included it other than too little usage?

Here's my extension method:

```csharp
		public static AndConstraint<ObjectAssertions> HaveFlag(this ObjectAssertions assertions, Enum flag, string because = "", params object[] reasonArgs)
		{
			Enum subject = assertions.Subject as Enum;

			if (subject == null)
				Assert.Fail("The provided object is not an enum. Type: {0}", assertions.Subject.GetType().Name);

			Execute.Assertion.BecauseOf(because, reasonArgs)
							 .ForCondition(subject.HasFlag(flag))
							 .FailWith("The enum was expected to have flag '{0}', but only has flags '{1}'.", flag, subject);
			
			return new AndConstraint<ObjectAssertions>(assertions);
		}
```
Thanks!

**Comment by dennisdoomen** [comment]
Sounds like a nice addition. You can speed up the integration if you could provide me with a pull request.

**Comment by AlexEndris** [comment]
Nice to hear. I'll do it later today. ;)

**Comment by dennisdoomen** [comment]
Fixed by #250
